src/renderer/components/DeleteWarnModal/index.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  Button,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { IDevice } from 'interfaces';
+import { memo, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { deleteDevice } from 'renderer/redux/ducks/devices';
+
+const DeleteWarnModal = ({ device }) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { t } = useTranslation('delete-warn');
+
+  const dispatch = useDispatch();
+
+  const handleClose = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  const handleRemoveDevice = useCallback(
+    (device: IDevice) => {
+      dispatch(deleteDevice(device));
+    },
+    [dispatch]
+  );
+
+  return (
+    <>
+      <Button size="sm" variant="ghost" colorScheme="red" onClick={onOpen}>
+        {t('delete')}
+      </Button>
+
+      <Modal isOpen={isOpen} onClose={handleClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>{t('warn')}</ModalHeader>
+          <ModalCloseButton />
+
+          <ModalFooter>
+            <Button mr={3} variant="ghost" onClick={handleClose}>
+              {t('cancel')}
+            </Button>
+            <Button
+              variant="outline"
+              onClick={() => handleRemoveDevice(device)}
```

Comment:
Because you don't need the device param anymore, you can just pass `handleRemoveDevice` directly

```suggestion
              onClick={handleRemoveDevice}
```